ivy/functional/frontends/jax/lax/control_flow_operators.py

```diff
@@ -58,3 +58,15 @@ def while_loop(cond_fun, body_fun, init_val):
     while cond_fun(val):
         val = body_fun(val)
     return val
+
+
+@to_ivy_arrays_and_back
+def scan(f, init, xs, length=None):
+    if xs is None:
+        xs = [None] * length
+    carry = init
+    ys = []
+    for x in xs:
+        carry, y = f(carry, x)
+        ys.append(y)
+    return carry, ivy.stack(ys)
```

Comment:
I'd recommend a good way of debugging this would be to create a python script which runs both the original jax.lax.scan function and your function for the same input. The return values for each function should be identical for all cases. But if you do this, do _not_ push the script to github.